### PR TITLE
Add some test coverage to allocating ENIs

### PR DESCRIPTION
**Pull request by mogren**
**What type of PR is this?**
cleanup

**Which issue does this PR fix**:
None

**What does this PR do**:
Add some unit test coverage for `awsutils.go`:
* Remove unused constants
* Actually test tagging
* Cover over-assigning secondary IPs

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
No issue

**Testing done on this change**:
Unit tests still pass

**Automation added to e2e**:
None

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:
No

**Does this PR introduce any user-facing change?**:
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/awsutils/awsutils_test.go [hunk 1]
```diff
@@ -605,29 +639,52 @@ func TestAllocIPAddresses(t *testing.T) {
...
+	err = ins.AllocIPAddresses(eniID, 0)
+	assert.NoError(t, err)
+}
+
+func TestAllocIPAddressesAlreadyFull(t *testing.T) {
+	ctrl, mockEC2 := setup(t)
+	defer ctrl.Finish()
+	// The required IP numbers(14) is the ENI's limit(14)
+	input := &ec2.AssignPrivateIpAddressesInput{
+		NetworkInterfaceId:             aws.String("eni-id"),
```
Comment (haouc) [inline pkg/awsutils/awsutils_test.go:679]: nit: should we use the consistent name "eniID"?

**Review by haouc — COMMENTED** [review]
Overall looks good to me. Just two nit.

File: pkg/awsutils/awsutils_test.go (hunk 1 above)
Comment (mogren) [reply pkg/awsutils/awsutils_test.go]: Thanks @haouc, addressed

**Review by jayanthvn — APPROVED** [review]
LGTM :)

**Pull request merged**